drivers/go-getting-started.md

```diff
@@ -75,13 +165,239 @@ book := Book{
     Title:   "ArangoDB Cookbook",
     NoPages: 257,
 }
+
 meta, err := col.CreateDocument(nil, book)
 if err != nil {
     // Handle error
 }
 fmt.Printf("Created document in collection '%s' in database '%s'\n", col.Name(), db.Name())
 ```
 
+Note that Go's `:=` operator declares and assigns with the automatic type of the function
+in one operation, so the type returned appear mysterious. It's also acceptable to
+declare variables explicitly using `var myvariable type` when learning these
+types. 
+Note also that Edge collections and Vertex collections use different methods
+from ordinary document collections, as they are contained by a Graph
+model and EdgeDefinitions.
+
+
+## Relationship between Go types and JSON
+
+A basic principle of the integration between Go and ArangoDB is the
+mapping from Go types to JSON documents. Data in the database map to
+types in Go through JSON. You will need at least two types in a Golang
+program to work with graphs
+
+Go uses a special syntax to map values like struct members like Key,
+Weight, Data, etc into JSON fields. Remember that member names should
+start with a capital letter to be accessible outside a packaged
+scope. You declare types and their JSON mappings once as in the
+examples below.
+
+```go
+// A typical document type
+
+type IntKeyValue struct {
+
+    Key    string  `json:"_key"`   // mandatory field (handle) - short name
+    Value  int     `json:"value"`
+}
+
+// A typical vertex type must have field matching _key
+
+type MyVertexNode struct {
+
+    Key     string    `json:"_key"` // mandatory field (handle) - short name
+ 
+      // other fields … e.g.
+
+    Data    string `json: "data"`   // Longer description or bulk string data
```

Comment:
The common practice is to use pointer fields for non-mandatory fields and adding `omitempty` annotation, so that you could update a single field without changing others. But I'm not sure if this is a getting-started topic.